grade-school/GradeSchoolTest.swift

```diff
@@ -2,82 +2,89 @@ import XCTest
         let expected = [2: ["Fred", "James", "Paul"]]
         XCTAssertEqual(Array(result.keys), Array(expected.keys))
-        XCTAssertEqual(result[2]!, expected[2]!)
+        XCTAssert(sameStudents(result[2], expected[2]))
     }
 
     func testAddStudentsToDifferentGrades() {
-        let school = GradeSchool()
-        school.addStudent("Chelsea", grade: 3)
+        var school = GradeSchool()
+        school.addStudent("Chelsea", grade:3)
         school.addStudent("Logan", grade: 7)
-        let result = school.db
+        let result = school.roster
         let expected = [3: ["Chelsea"], 7: ["Logan"]]
         XCTAssertEqual(Array(result.keys).sort(>), Array(expected.keys).sort(>))
-        XCTAssertEqual(result[3]!, expected[3]!)
+        XCTAssert(sameStudents(result[3], expected[3]))
     }
 
     func testGetStudentsInAGrade() {
-        let school = GradeSchool()
+        var school = GradeSchool()
         school.addStudent("Franklin", grade: 5)
         school.addStudent("Bradley", grade: 5)
         school.addStudent("Jeff", grade: 1)
-        let result = school.studentsInGrade(5)
-        let expected = [ "Franklin", "Bradley" ]
-        XCTAssertEqual(result, expected)
+        let result   = school.studentsInGrade(5)
+        let expected = ["Franklin", "Bradley"]
+        XCTAssert(sameStudents(result, expected))
     }
 
     func testGetStudentsInANonExistantGrade() {
         let school = GradeSchool()
         let result = school.studentsInGrade(1)
-    
-        let expected: [String] = []
-        XCTAssertEqual(result, expected)
+
+        let expected = [String]()
+        XCTAssert(sameStudents(result, expected))
     }
 
     func testSortSchool() {
-        let school = GradeSchool()
-        
-        school.addStudent("Jennifer", grade:4)
-        school.addStudent("Kareem", grade:6)
-        school.addStudent("Christopher", grade:4)
+        var school = GradeSchool()
+        school.addStudent("Jennifer", grade: 4)
+        school.addStudent("Kareem", grade: 6)
+        school.addStudent("Christopher", grade: 4)
         school.addStudent("Kyle", grade: 3)
-        let result = school.sortedRoster()
-        
+        let result = school.sortedRoster
+
         let expected = [
             3 : ["Kyle"],
-            4 : [ "Christopher", "Jennifer" ],
-            6 : [ "Kareem"]
+            4 : ["Christopher", "Jennifer"],
+            6 : ["Kareem"]
         ]
-        
+
         XCTAssertEqual(Array(result.keys).sort(>), Array(expected.keys).sort(>))
-        XCTAssertEqual(result[3]!, expected[3]!)
-        XCTAssertEqual(result[4]!, expected[4]!)
-        XCTAssertEqual(result[6]!, expected[6]!)
+        XCTAssert(sameStudents(result[3], expected[3]))
+        XCTAssert(sameStudents(result[4], expected[4]))
+        XCTAssert(sameStudents(result[6], expected[6]))
+    }
+
+    func sameStudents<C: CollectionType where C.Generator.Element == String>(result: C?, _ expected: [String]?) -> Bool {
```

Comment:
I don't like introducing this function here.  xCTAsssertEquals should be able to test optional. I rather remove the unwrapping than introducing this function. You may need to implement the equality protocol in the example file.